nmi_interface.cpp

```diff
@@ -33,25 +35,52 @@ NMI::NMI(sdbusplus::bus::bus& bus, const char* path) :
 void NMI::nMI()
 {
     using namespace phosphor::logging;
-    using sdbusplus::exception::SdBusError;
     using InternalFailure =
         sdbusplus::xyz::openbmc_project::Common::Error::InternalFailure;
+    struct pdbg_target *procTarget, *childTarget, *coreTarget;
+    char path[16];
+
+    // Find and probe the right child target (PIB) before invoking
+    // stop all chip op.
+    pdbg_for_each_class_target("proc", procTarget) {
+
+        sprintf(path,
+                "/%s%d/%s","proc", pdbg_target_index(procTarget),
+                "pib");
+
+        childTarget = pdbg_target_from_path(NULL, path);
+
+        //Bail out if child target not available.
+        if (childTarget == NULL) {
+            log<level::ERR>("No proc target found");
+            report<InternalFailure>();
+            return;
+        }
 
-    constexpr auto SYSTEMD_SERVICE = "org.freedesktop.systemd1";
-    constexpr auto SYSTEMD_OBJ_PATH = "/org/freedesktop/systemd1";
-    constexpr auto SYSTEMD_INTERFACE = "org.freedesktop.systemd1.Manager";
+        // Probe the child (pib) target
+        if(pdbg_target_probe(childTarget) != PDBG_TARGET_ENABLED)
+            continue;
+    
+        //We are probing the cores to keep the special wakeup asserted.  
```

Comment:
Not required  this special wake-up here  As discussed , pdbg should force the  special wkae-up assert flag to indicate sbe to do soecial wake-up prior to stop/sreset .

Reply:
Done